src/dos/drive_fat.cpp

```diff
@@ -1284,23 +1306,31 @@ bool fatDrive::FindNextInternal(uint32_t dirClustNumber, DOS_DTA &dta, direntry
 	trimString(&find_name[0], sizeof(find_name));
 	trimString(&extension[0], sizeof(extension));
 	
-	//if(!(sectbuf[entryoffset].attrib & DOS_ATTR_DIRECTORY))
+	//if(!(sectbuf[entryoffset].attrib & FatAttributeDirectory._data))
 	if (extension[0]!=0) {
 		safe_strcat(find_name, ".");
 		safe_strcat(find_name, extension);
 	}
 
 	/* Compare attributes to search attributes */
-
-	//TODO What about attrs = DOS_ATTR_VOLUME|DOS_ATTR_DIRECTORY ?
-	if (attrs == DOS_ATTR_VOLUME) {
-		if (!(sectbuf[entryoffset].attrib & DOS_ATTR_VOLUME)) goto nextfile;
+	FatAttributeFlags attr_mask = {};
+	attr_mask.directory = true;
+	attr_mask.volume    = true;
+	attr_mask.system    = true;
+	attr_mask.hidden    = true;
+
+	// TODO What about volume/directory attributes?
+	if (attrs == FatAttributeVolume) {
+		if (!(sectbuf[entryoffset].attrib & FatAttributeVolume._data)) {
+			goto nextfile;
+		}
 		dirCache.SetLabel(find_name, false, true);
 	} else {
-		if (~attrs & sectbuf[entryoffset].attrib & (DOS_ATTR_DIRECTORY | DOS_ATTR_VOLUME | DOS_ATTR_SYSTEM | DOS_ATTR_HIDDEN) ) goto nextfile;
+		if (~(attrs._data) & sectbuf[entryoffset].attrib & attr_mask._data) {
+			goto nextfile;
+		}
 	}
```

Comment:
I think we can levrage FatAttributeFlags to more directly test these volume, directory, etc.. bits instead of using bit-wise operations.

Push an example here: a17542b37

Reply:
That would be preferable, yes. We should try to get as much use out of named bitfields as we can; it greatly enhances readability and reduces the likelihood of bit-twiddling bugs.

Reply:
OK, cherry-picked your change.